titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,401 @@
+        PixelSelectionMethod.first, description="Pixel selection method."
+    ),
+    image_params: CommonTileParams = Depends(),
+    mosaic_params: CommonMosaicParams = Depends(),
+):
+    """Read MosaicJSON tile"""
+    # TODO: Maybe use ``read_mosaic`` defined above (depending on cache behavior which is still TBD)
+    pixsel = pixel_selection.method()
+    timings = []
+    headers: Dict[str, str] = {}
+
+    with utils.Timer() as t:
+        with MosaicBackend(mosaic_params.mosaic_path) as mosaic:
+            assets = mosaic.tile(x=x, y=y, z=z)
+            if not assets:
+                raise TileNotFoundError(f"No assets found for tile {z}/{x}/{y}")
+    timings.append(("Read-mosaic", t.elapsed))
+
+    tilesize = 256 * scale
+
+    # Rio-tiler-mosaic uses an external ThreadPoolExecutor to process multiple assets at once but we want to use the
+    # executor provided by the event loop.  Instead of calling ``rio_tiler_mosaic.mosaic.mosaic_tiler`` directly we will
+    # transcribe the code here and use the executor provided by the event loop.  This also means we define this function
+    # as a coroutine (even though nothing that is called is a coroutine), since the event loop's executor isn't
+    # available in normal ``def`` functions.
+    # https://github.com/cogeotiff/rio-tiler-mosaic/blob/master/rio_tiler_mosaic/mosaic.py#L37-L102
+    _tiler = partial(
+        cogeoTiler,
+        tile_x=x,
+        tile_y=y,
+        tile_z=z,
+        tilesize=tilesize,
+        indexes=image_params.indexes,
+        # expression=image_params.expression, # TODO: Figure out why expression kwarg doesn't work
+        nodata=image_params.nodata,
+        **image_params.kwargs,
+    )
+    futures = [run_in_threadpool(_tiler, asset) for asset in assets]
+
+    semaphore = asyncio.Semaphore(int(os.getenv("MOSAIC_CONCURRENCY", 10)))
+
+    with utils.Timer() as t:
+        async with semaphore:
+            for fut in asyncio.as_completed(futures):
+                try:
+                    tile, mask = await fut
+                except Exception:
+                    # Gracefully handle exceptions
+                    continue
+
+                tile = numpy.ma.array(tile)
+                tile.mask = mask == 0
+                pixsel.feed(tile)
+                if pixsel.is_done:
+                    break
+    timings.append(("Read-tiles", t.elapsed))
+
+    tile, mask = pixsel.data
+    if tile is None:
+        raise TileNotFoundError(f"Tile {z}/{x}/{y} was not found")
+
+    if not format:
+        format = ImageType.jpg if mask.all() else ImageType.png
+
+    with utils.Timer() as t:
+        tile = utils.postprocess(
+            tile,
+            mask,
+            rescale=image_params.rescale,
+            color_formula=image_params.color_formula,
+        )
+    timings.append(("Post-process", t.elapsed))
+
+    bounds = mercantile.xy_bounds(mercantile.Tile(x, y, z))
+    dst_transform = from_bounds(*bounds, tilesize, tilesize)
```

Comment:
both rio-tiler and rio-tiler-crs have a function for this:
https://github.com/cogeotiff/rio-tiler/blob/cb0a709144b4589bb9940e56716eb741a53f71fd/rio_tiler/utils.py#L315
https://github.com/cogeotiff/rio-tiler-crs/blob/master/rio_tiler_crs/reader.py#L24-L34

Reply:
https://github.com/developmentseed/titiler/pull/40/commits/52b416d61244e3cb02ea7b68cad252a3387de2b2